qndiag/qndiag.py

```diff
@@ -74,6 +78,20 @@ def qndiag(C, B0=None, weights=None, max_iter=1000, tol=1e-6,
     https://arxiv.org/abs/1811.11433
     """
     t0 = time()
+    if not isinstance(C, np.ndarray):
+        raise TypeError('Input tensor C should be a numpy array.')
+    if C.ndim != 3:
+        raise ValueError('Input tensor C should have 3 dimensions')
+    if C.shape[1] != C.shape[2]:
+        raise ValueError('The last two dimensions of C should be the same')
+    if check_sympos:
+        if not np.allclose(C, C.swapaxes(-1, -2)):
+            raise ValueError('C does not contain only symmetric matrices')
+        try:  # Check positivity using Cholesky
+            np.linalg.cholesky(C)
+        except np.linalg.LinAlgError:
+            raise ValueError('C contains some non positive matrices') from None
```

Comment:
why the `from None` ?

Reply:
weird hack found here: https://stackoverflow.com/questions/52725278/during-handling-of-the-above-exception-another-exception-occurred

without it, it prints two exceptions, first the LinAlgError and then the ValueError